### DISCUSSION TITLE
Visual Block Mode and increment

### DISCUSSION THREAD

**Discussion by wf5w** [neovim/neovim#31285] [General]
Hello a newbie with neovim here.

In vim, I am used to doing a block mode increment. CTRL-V then select the column, and then do a g^a to increment say from 01 to 09

the increment happens like it is supposed to, but
the selection block doesn't show up

Not sure what needs to be configured to do this.

I have neovim 0.10.2 and also used the neovim kickstarter to configure my neovim.

Thanks, 
Jerry

**Comment by zeertzjq** [comment]
What do you mean by "the selection block doesn't show up"? If you want to restore the selection block, you can press `gv`.

**Reply by wf5w** [reply]
well, in vim, when I do the visual-block mode, what I select is highlighted. 

when I do the same in neovim, nothing is highlighted but it still works. So there is no visual feedback of what I selected.

**Reply by zeertzjq** [reply]
What terminal are you using?

**Reply by wf5w** [reply]
kitty  0.37.0
let me try gnome terminal and see if I get the same thing.

**Reply by wf5w** [reply]
Sure enough! kitty is the culprit! It worked as it should have in gnome terminal.

Now to track down why in kitty.

**Reply by wf5w** [reply]
Actually, it isn't kitty at all.
1. vim - gnome terminal - works fine
2. nvim - gnome terminal - works fine
3. vim - kitty - works fine
4. nvim - kitty - if it works I can't tell it.

I suspect it is something to do with neovim and kitty. not sure what.

Is there a setting in neovim to make the highlight stand out more? that could be the problem.

**Reply by wf5w** [reply]
ok. After further investigation. It appears to be the dark theme I am using.

I did a kitten themes command

I have tried a half a dozen dark themes and they all exhibit the same behavior. They don't highlight.

I tried the very first light theme, and highlighting works fine and as expected.

I *DO* prefer a dark theme BUT ...

Anyone know of a dark theme that highlights better?

**Reply by zeertzjq** [reply]
Hmm, what's you Nvim version? Does it work if you change the value of the `'termguicolors'` option?

**Reply by wf5w** [reply]
that was in my original post, but it is 0.10.2

**Reply by wf5w** [reply]
in my init.lua, I did a vim.o.termguicolors = true

it did nothing obvious.

**Reply by zeertzjq** [reply]
Hmm, that's really strange, as I've never encountered such problems in kitty.

**Reply by wf5w** [reply]
I have tried a dozen different dark themes. all the same.

I now believe that it IS highlighting, but it is sooooooooo light that to my tired eyes it may as well be non-existent.

**Comment by wf5w** [comment]
ok. Did hours worth of searching.

I found this: in the vim group:

If I do this in command mode

:hi Visual  guifg=#000000 guibg=#FFFFFF gui=none

It is very visible.

This is what I want.

Now, how do I set this automatically in init.lua?

**Reply by wf5w** [reply]
ok. This worked. All of this was really hard to figure out!

put this in the init.lua

vim.cmd [[hi Visual guifg=#000000 guibg=#FFFFFF gui=none]]
